Sources/RSocketCore/Channel Handler/DemultiplexerHandler.swift

```diff
@@ -40,7 +41,10 @@ internal struct DemultiplexerRouter {
     
     internal var connectionSide: ConnectionRole
     
-    internal func route(for streamId: StreamID) -> Route {
+    internal func route(for streamId: StreamID, type: FrameType) -> Route {
+        guard type != .metadataPush else {
+            return .responder
+        }
         switch (streamId.generatedBy, connectionSide) {
         case (nil, _):
             return .connection
```

Comment:
can we move the type check inside the `.connection` case? 

```suggestion
        switch (streamId.generatedBy, connectionSide) {
        case (nil, _):    
            guard type != .metadataPush else {
                return .responder
            }
            return .connection
```